### PR TITLE
Remove explicit interface classes

### PR DESCRIPTION
**Pull request by scarvill91**
You may also want to check out the last commit or two to the master branch before this. I had started working on this story before you asked me to group changes as pull requests.

### CODE REVIEW

File: lib/player_factory.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+require_relative 'human_player'
+require_relative 'computer_player'
+
+module TicTacToe
+  module PlayerFactory
+    def self.build(config)
+      shared_parameters = {player_mark: config[:player_mark]}
+
+      case config[:type]
+      when :human
```
Comment (ericmeyer) [inline lib/player_factory.rb]: I would use constants for these sort of values (`:human`, `:computer`). Right now, they are likely duplicated in few places and are what are referred to as "Magic Strings."

File: spec/computer_player_spec.rb [hunk 2]
```diff
@@ -14,12 +14,6 @@ module TicTacToe
...
     let(:neg_infinity) { -Float::INFINITY }
 
-    it 'implements methods required by Player interface' do
-      Player.required_methods.each do |method|
-        expect(ai).to respond_to(method)
-      end
-    end
-
     describe '#initialize' do
       it 'takes a parameters hash with the player\'s mark, opponent\'s mark, and the game board' do
```
Comment (ericmeyer) [inline spec/computer_player_spec.rb]: I would use double quotes for the `it` string. That way, you don't have to escape the single quotes.

In general, I like to favor using double quotes. There are a few main reasons. Keep in mind that String interpolation requires double quotes:

1. I don't have to change single quotes to double quotes if I want to start interpolating where I was not before.
2. I don't have a mixture of single quotes and double quotes. I like consistency.
3. Related to 2., I don't have to think about which quote to use.

File: spec/game_spec.rb [hunk 3]
```diff
@@ -12,10 +12,11 @@ module TicTacToe
       end
 
       context 'when given parameters' do
-        it 'creates a board of the given size' do
-          custom_game = described_class.new(board_size: 4)
+        it 'uses given board' do
```
Comment (ericmeyer) [inline spec/game_spec.rb]: Slightly awkward wording. Maybe try `"uses the given board"`

Comment (scarvill91) [inline spec/game_spec.rb]: Done.

File: spec/computer_player_spec.rb (hunk 2 above)
Comment (scarvill91) [inline spec/computer_player_spec.rb]: Thanks - will do.

File: lib/player_factory.rb (hunk 1 above)
Comment (scarvill91) [inline lib/player_factory.rb]: OK - I'll do that

**Pull request merged**